.claude/hooks/check-query-safety.sh

```
@@ -0,0 +1,45 @@
+
+# Only check TypeScript files
+echo "$FILE_PATH" | grep -qE '\.(ts|tsx)$' || exit 0
+
+# Get the content being written/edited
+NEW_CONTENT=$(echo "$INPUT" | jq -r '.tool_input.new_string // .tool_input.content // empty')
+[ -z "$NEW_CONTENT" ] && exit 0
+
+# Detect template literals with interpolation that look like queries
+# Check for SQL/Cypher keywords near ${...} interpolation
+QUERY_KEYWORDS='(SELECT|INSERT|UPDATE|DELETE|MERGE|MATCH|CREATE|DROP|ALTER|CALL|RETURN|WITH|UNWIND)'
+if echo "$NEW_CONTENT" | grep -qiE "${QUERY_KEYWORDS}" && echo "$NEW_CONTENT" | grep -qF '${'; then
```

Shellcheck (0.11.0)

[info] Expressions don't expand in single quotes, use double quotes for that.

(SC2016)